### ISSUE TITLE
a11y: Clear filters link is not keyboard focusable

### ISSUE THREAD

**Issue by iamjolly** [18F/all_sorns#136]
## Description:
The "clear section filters" anchor is missing an `href` attribute and corresponding value. Browsers interpret anchors without `href`s as "placeholder" links that are omitted from the browser's focus order. They are not navigable by the keyboard, and sighted keyboard-only users are most impacted by this barrier to access. Screen reader users are also impacted, as placeholder links may not be announced as links, so this may cause some confusion for them. Only sighted mouse users can reliably use this functionality in its current state.

## Screenshot:
![screenshot showing anchor without href](https://user-images.githubusercontent.com/1093423/105251276-68d4ba00-5b38-11eb-838b-e625b2f43af5.png)

## WCAG SC
[Success Criterion 2.1.1 Keyboard (Level A)](https://www.w3.org/WAI/WCAG21/Understanding/keyboard.html): All functionality of the content is operable through a keyboard interface without requiring specific timings for individual keystrokes, except where the underlying function requires input that depends on the path of the user's movement and not just the endpoints.

### Additional guidance:
- [Mozilla MDN: The anchor element, onclick events](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#onclick_events) says: 
> Anchor elements are often abused as fake buttons by setting their href to # or javascript:void(0) to prevent the page from refreshing, then listening for their click events .
> 
> These bogus href values cause unexpected behavior when copying/dragging links, opening links in a new tab/window, bookmarking, or when JavaScript is loading, errors, or is disabled. They also convey incorrect semantics to assistive technologies, like screen readers.
> 
> Use a `<button>` instead. In general, you should only use a hyperlink for navigation to a real URL.
- [Scott O'Hara article _The accessibility of placeholder links_](https://www.scottohara.me/note/2019/07/17/placeholder-link.html)

## Location of code:
[/app/views/sorns/_sections.html.erb, Line 6](https://github.com/18F/all_sorns/blob/41c4787de09eff4ff2c745250df070efa14504e2/app/views/sorns/_sections.html.erb#L6)

## Recommended fix:
- *Best option:* Use [the unstyled `button` component from USWDS](https://designsystem.digital.gov/components/button/) instead of the `a` element, or...
- *Acceptable option:* provide an `href` attribute with value that allows the link to be focusable and actionable for keyboard-only users.
